docs/advanced/document-store/redis.md

```diff
@@ -0,0 +1,334 @@
+| `port`            | Port of the Redis Server                                                                          | `6379`                                            |
+| `redis_config`    | Other Redis configs in a Dict and pass to `Redis` client constructor, e.g. `socket_timeout`, `ssl`| `{}`                                              |
+| `index_name`      | Redis index name; the name of RedisSearch index to set this DocumentArray                         | `'idx'`                                           |
+| `n_dim`           | Dimensionality of the embeddings                                                                  | `None`                                            |
+| `flush`           | Boolean flag indicating whether to clear previous DocumentArray in Redis                          | `False`                                           |
+| `update_schema`   | Boolean flag indicating whether to update Redis Search schema                                     | `True`                                            |
+| `distance`        | Similarity distance metric in Redis                                                               | `'COSINE'`                                        |
+| `batch_size`      | Batch size used to handle storage updates                                                         | `64`                                              |
+| `method`          | Vector similarity index algorithm in Redis                                                        | `'HNSW'`                                          |
+| `ef_construction` | Optional parameter for Redis HNSW algorithm                                                       | `200`                                             |
+| `m`               | Optional parameter for Redis HNSW algorithm                                                       | `16`                                              |
+| `ef_runtime`      | Optional parameter for Redis HNSW algorithm                                                       | `10`                                              |
+| `block_size`      | Optional parameter for Redis FLAT algorithm                                                       | `1048576`                                         |
+| `initial_cap`     | Optional parameter for Redis HNSW and FLAT algorithm                                              | `None`, defaults to the default value in Redis    |
+| `columns`         | Other fields to store in Document and build schema                                                | `None`                                            |
+
+You can check the default values in [the docarray source code](https://github.com/jina-ai/docarray/blob/main/docarray/array/storage/redis/backend.py)
+
+
+```{note}
+This Document Store will support storing multiple DocumentArrays and full-text search soon.
+```
```

Comment:
don't forget the geo filters and additional features are not supported yet

Reply:
wait, `This Document Store will support storing multiple DocumentArrays` is against the design principle of a DocumentArray. Note that `da = DocumentArray(storage='redis')` is after all **one** DocumentArray. By concept-substitution, you see the logic problem here: it doesn't make sense to say "This DocumentArray will support storing multiple DocumentArrays"